server/bastionlab_polars/src/lib.rs

```diff
@@ -86,6 +90,53 @@ impl BastionLabPolars {
         }
     }
 
+    fn update_log(
+        self: &BastionLabPolars,
+        new_row: DataFrame,
+        ) -> Result<(), Status> {
+        let mut dfs = self.dataframes.write().unwrap(); 
+        let artifact = dfs.get_mut(LOG_DF);
+        match artifact{
+            Some(artifact) => { //log dataframe exists, append new_row and save
+                artifact.dataframe = artifact.dataframe.vstack(&new_row).unwrap_or(artifact.dataframe.clone());
+                drop(dfs);
```

Comment:
Usually you don't need to explicitly drop something in Rust.
Why is this needed here? If this is just a matter of scope, use a block {}